lib/facility.rb

```diff
@@ -1,14 +1,64 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
+  attr_reader :name, :address, :phone,
+              :services, :registered_vehicles, :collected_fees
+              :administer_written_test
+  def initialize(name:, address:, phone:)
     @name = name
     @address = address
     @phone = phone
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
+    @administer_written_test
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def register_vehicle(vehicle)
+    return nil unless services.include?("Vehicle Registration")
+    vehicle.register
+    vehicle.registration_date == Date.today
+    @collected_fees += fee_amount(vehicle)
```

Comment:
Great use of a helper method! I will peek in the spec file to see if there is a test for the helper method. :)

Reply:
I did not see a test. Remember to test all methods that are in your class, even helper methods.